### PR TITLE
Backport #1812 to release 2020.06

### PR DESCRIPTION
**Pull request by avnishn**
* Flatten observations if necessary

This commit takes care to convert observations that belong to
spaces that aren't box spaces into observations that would
come from box spaces via akro observation space flattening.

It is really a temporary change. The better solution would be
to modify our GymEnv/GarageEnv and BulletEnv Wrappers to flatten
observations spaces that aren't box spaces.

This is a smaller change, however, the HER replay buffer would
need to be rewritten, which is a much larger change/refactor.

I have refrained from modifying the primitives that belong to
multi task and metalearning algorithms specifically.

Additionally, this change updates usages of ray to reflect
the update in parameter names from ray 0.8.6->0.8.7

### CODE REVIEW

**Comment by avnishn** [comment]
I think I may have accidentally added some docs from after the release, but I'm not sure if this matters.

**Comment by avnishn** [comment]
@gitanshu @ryanjulian, the CI fails because its using a docker container that has been built with an older version of ray. The newest version of ray changes the parameter name `use_webui` to `use_dashboard` for the function `ray.init`. We actually never pinned ray in 2020.06, so I figured that I should also update this parameter name in my backport, however the CI seems to be using this old ray. Can you please advise on next steps.

**Comment by ryanjulian** [comment]
The CI builds the docker container from scratch: https://travis-ci.com/github/rlworkgroup/garage/jobs/380405992

Could it be that the new version of ray does not support Python 3.5, or that some other dependency is holding ray back?

**Comment by ryanjulian** [comment]
For a release branch, it's probably easier just to pin ray down to your preferred version, i.e.e `ray<=0.8.6`

File: docs/user/meta_multi_task_rl_exp.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
+# Run Meta-/Multi-Task RL Experiments
```
Comment (ryanjulian) [inline docs/user/meta_multi_task_rl_exp.md:1]: please do not update the documentation

**Review by ryanjulian — COMMENTED** [review]
this has tons of merge conflicts and changes to unrelated files. please let me know when it's ready for recview.

**Comment by avnishn** [comment]
Please see #1997. I had issues with this branch because branches that start with release- are treated as protected branches.

**Pull request closed without merging**